eureka-client/src/test/java/com/netflix/discovery/converters/wrappers/CodecWrappersTest.java

```diff
@@ -0,0 +1,61 @@
+package com.netflix.discovery.converters.wrappers;
+
+import junit.framework.Assert;
+import org.junit.Test;
+
+import javax.ws.rs.core.MediaType;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * @author David Liu
+ */
+public class CodecWrappersTest {
+
+    private static String testWrapperName = "FOO_WRAPPER";
+
+    @Test
+    public void testRegisterNewWrapper() {
+        Assert.assertNull(CodecWrappers.getEncoder(testWrapperName));
+        Assert.assertNull(CodecWrappers.getDecoder(testWrapperName));
+
+        CodecWrappers.registerWrapper(new TestWrapper());
+
+        Assert.assertNotNull(CodecWrappers.getEncoder(testWrapperName));
+        Assert.assertNotNull(CodecWrappers.getDecoder(testWrapperName));
+    }
+
+    private final class TestWrapper implements EncoderDecoderWrapper {
```

Comment:
I'm curious what the purpose of supporting multiple codecs is.  Is it to be less opinionated about dependencies provided to downstreams?  If so, could you pick one and shade it instead?